lib/vif-plug-providers/representor/vif-plug-representor.c

```diff
@@ -42,6 +43,347 @@ static struct shash devlink_ports;
 
 /* Max number of physical ports connected to a single NIC SoC. */
 #define MAX_NIC_PHY_PORTS 64
+
+struct port_node {
+    struct hmap_node mac_vf_node;
+    struct hmap_node ifindex_node;
+    struct hmap_node bus_dev_node;
+    uint32_t netdev_ifindex;
+    char *netdev_name;
+    uint32_t number;
```

Comment:
Based on https://github.com/torvalds/linux/blob/0c947b893d69231a9add855939da7c66237ab44f/include/net/devlink.h#L92-L94 there are different attributes for different flavors:

```
	union {
		struct devlink_port_phys_attrs phys;
		struct devlink_port_pci_pf_attrs pci_pf;
		struct devlink_port_pci_vf_attrs pci_vf;
		struct devlink_port_pci_sf_attrs pci_sf;
	};
```

While `index` is a part of `struct devlink_port` which contains `struct devlink_port_attrs`:
https://github.com/torvalds/linux/blob/0c947b893d69231a9add855939da7c66237ab44f/include/net/devlink.h#L123

In case of `PHYSICAL` ports we are expecting a 32-bit port number ("split group"):
https://github.com/torvalds/linux/blob/0c947b893d69231a9add855939da7c66237ab44f/include/net/devlink.h#L26-L30
```
struct devlink_port_phys_attrs {
	u32 port_number; /* Same value as "split group".
			  * A physical port which is visible to the user
			  * for a given port flavour.
			  */
```

In case of PF representors (`PCI_PF`) we expecting a 16-bit PF number as an attribute:

https://github.com/torvalds/linux/blob/0c947b893d69231a9add855939da7c66237ab44f/include/net/devlink.h#L42

So it appears to be that the port index is an identifier for devlink ports in the devlink port subsystem while the physical port number is a different thing.

Reply:
Yes, for the internal storage we use a single variable which can hold any of them + a flavor to determine which it is. The functions used to interface with the data makes sure the appropriate value is retrieved/stored.

I'd be happy to add comments and / or changing names of the variables to make this more clear.

Reply:
Done